velox/functions/prestosql/GeometryFunctions.h

```diff
@@ -391,4 +417,53 @@ struct StUnionFunction {
   }
 };
 
+// Accessors
+template <typename T>
+struct StXFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<double>& result,
+      const arg_type<Geometry>& geometry) {
+    std::unique_ptr<geos::geom::Geometry> geosGeometry =
+        geospatial::deserializeGeometry(geometry);
+    if (geosGeometry->getGeometryTypeId() !=
+        geos::geom::GeometryTypeId::GEOS_POINT) {
+      throw Status::UserError(fmt::format(
+          "ST_X requires a Point geometry, found {}",
+          geosGeometry->getGeometryType()));
+    }
+    if (geosGeometry->isEmpty()) {
+      return false;
+    }
+    auto coordinate = geosGeometry->getCoordinate();
+    result = coordinate->x;
+    return true;
+  }
+};
+
+template <typename T>
+struct StYFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<double>& result,
+      const arg_type<Geometry>& geometry) {
+    std::unique_ptr<geos::geom::Geometry> geosGeometry =
+        geospatial::deserializeGeometry(geometry);
+    if (geosGeometry->getGeometryTypeId() !=
+        geos::geom::GeometryTypeId::GEOS_POINT) {
+      throw Status::UserError(fmt::format(
+          "ST_Y requires a Point geometry, found {}",
+          geosGeometry->getGeometryType()));
+    }
+    if (geosGeometry->isEmpty()) {
```

Comment:
i noticed a test case with input x coordinate is null, would that exercise this code path? If not, can you please add one?

Reply:
Good catch.  I'll add explicit tests for empty points.